Orange/widgets/data/owpaintdata.py

```diff
@@ -1154,20 +1158,20 @@ def _on_editing_finished(self):
         self.undo_stack.endMacro()
 
     def execute(self, command):
-        if isinstance(command, (Append, DeleteIndices, Insert, Move)):
-            if isinstance(command, (DeleteIndices, Insert)):
-                self._selected_indices = None
+        assert isinstance(command, (Append, DeleteIndices, Insert, Move)), \
+            "Non normalized command"
+        if isinstance(command, (DeleteIndices, Insert)):
+            self._selected_indices = None
 
-                if isinstance(self.current_tool, SelectTool):
```

Comment:
@ales-erjavec, should this `if` really be nested? Am I overlooking something or is this condition always false?

If resetting the tool is redundant, `_reset()` method can remain private.

Reply:
Yes the condition can be true and should be nested.

Reply:
Oops, apologies for asking. I was so sloppy that I overlooked that this one checks `self.current_tool`, not `command`, so I didn't understand why is this needed. :(